tests/misc/sys_settrace_features.py

```diff
@@ -21,7 +21,7 @@ def print_stacktrace(frame, level=0):
             frame.f_globals["__name__"],
             frame.f_code.co_name,
             # reduce full path to some pseudo-relative
-            "misc" + "".join(frame.f_code.co_filename.split("tests/misc")[-1:]),
+            "misc" + "".join(frame.f_code.co_filename.replace("\\", "/").split("tests/misc")[-1:]),
```

Comment:
how about:
```python
"sys_settrace_" + frame.f_code.co_filename.split("sys_settrace_")[-1],
```
that should work for everything, even `./run-tests -d misc`?

Reply:
It would work for this file, but will still fail for the other 2 tests

Reply:
I tried it on the other 2 and it passes (and they can be run from any dir it seems) if the .exp files are also updated.

Eg, from top-level of repo:
```
MICROPY_MICROPYTHON=./ports/unix/micropython-dev ./tests/run-tests -d tests/misc
```
(that's not a very common way to run it, I admit)

Reply:
Actually I was just hesitating to change the .exp files since they are so large. But it's also bit silly to use the rather complicated "misc" + "".join(....) when it's just a matter of getting the filename out, and having the meaningless "miscmisc" in the tests.

Note what you propose still needs to replace slashes on Windows for the tests in the sys_settrace_subdir.
So: can we settle on
```
"sys_settrace_" + frame.f_code.co_filename.split("sys_settrace_")[-1].replace('\\', '/'),
```
in all 3 files, plus removal of `miscmisc/` from the current .exp files?
Alternatively the slash replacement can be left out of the loop/generator tests since they are not affected.

I do often run tests like `./run-tests -d ./misc` in order not to have to wait for the full test suite to complete.

Reply:
Other alternative: sine all our files have `trace_` in the name this would work for all 3 tests:
```
"trace_" + frame.f_code.co_filename.split("trace_")[-1]
```
but sort of obfuscates the output.

Reply:
> Actually I was just hesitating to change the .exp files since they are so large. But it's also bit silly to use the rather complicated "misc" + "".join(....) when it's just a matter of getting the filename out, and having the meaningless "miscmisc" in the tests.

Changing the .exp files is ok, I think it's a worthy change to clean them up (the miscmisc is certainly not useful).

I think the reason the .exp's are needed is because MicroPython has different line numbers for while loops due to optimisation of the bytecode.

> Note what you propose still needs to replace slashes on Windows for the tests in the sys_settrace_subdir.

Does it?  Hmm... I thought that the sys_settrace_features would work b/c running CPython on the host (eg Windows) would produce the correct slash to match.

> I do often run tests like `./run-tests -d ./misc` in order not to have to wait for the full test suite to complete.

Yes I do that all the time, but use `./run-tests -d misc` instead (which I think is still broken for these settrace tests).



> So: can we settle on
> 
> ```
> "sys_settrace_" + frame.f_code.co_filename.split("sys_settrace_")[-1].replace('\\', '/'),
> ```
> 
> in all 3 files, plus removal of `miscmisc/` from the current .exp files?
> Alternatively the slash replacement can be left out of the loop/generator tests since they are not affected.

That sounds good to me, including leaving out the slash replacement in the loop and generator tests.

Reply:
> I thought that the sys_settrace_features would work b/c running CPython on the host (eg Windows) would produce the correct slash to match.

CPython will produce sys_settrace_subdir\trace_generic.py whereas MicroPython will always produce forward slashes.

>  (which I think is still broken for these settrace tests).

No it will work now, since the entire 'problematic' part comes before the filename and is skipped.

Just to doublecheck: did you see my previous comment with other alternatives (could have been I posted that while you were writing your answer)? The one with using `"sys_settrace_" + frame.f_code.co_filename.split("sys_settrace_")[-1]` everywhere and renaming the 2 files in sys_settrace_subdir is perhaps the best after all.

Reply:
> > (which I think is still broken for these settrace tests).
> 
> No it will work now, since the entire 'problematic' part comes before the filename and is skipped.

Ah, I meant "which I think is currently/also broken for these settrace tests", but will be fixed by this PR.

> Just to doublecheck: did you see my previous comment with other alternatives

Yes I did see it.

> The one with using `"sys_settrace_" + frame.f_code.co_filename.split("sys_settrace_")[-1]` everywhere and renaming the 2 files in sys_settrace_subdir is perhaps the best after all.

If you like this approach feel free to do it that way, I'm fine with it (and it is perhaps best because there are no slashes to worry about).

Reply:
Ok, took the last path so the code remains simple and the same for all files.